samples/set_best_video_mode/main.cpp

```diff
@@ -0,0 +1,70 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <windows.h>
+#include <vector>
+
+// Screen dimension globals
+static int SCREEN_WIDTH = 640;
+static int SCREEN_HEIGHT = 480;
+
+int main(void)
+{
+    // set the default video mode
+    XVideoSetMode(SCREEN_WIDTH, SCREEN_HEIGHT, 32, REFRESH_DEFAULT);
+
+    // create an empty list for the available video modes
+    std::vector<VIDEO_MODE> availableVideoModes;
+
+    // save the best video mode here
+    VIDEO_MODE bestVideoMode = {0, 0, 0, 0};
+
+    VIDEO_MODE vm;
+    int bpp = 32;  // Bits per pixel
+    void *p = NULL;  // Initialize to NULL for the first call
+
+    // get the available video modes
+    while (XVideoListModes(&vm, bpp, REFRESH_DEFAULT, &p)) {
+        availableVideoModes.push_back(vm);
+
+        // ensure height is equal to or better than the current best video mode
+        if (vm.height < bestVideoMode.height) {
+            continue;
+        }
+
+        // ensure width is equal to or better than the current best video mode
+        if (vm.width < bestVideoMode.width) {
+            continue;
+        }
+
+        // ensure bpp is equal to or better than the current best video mode
+        if (vm.bpp < bestVideoMode.bpp) {
+            continue;
+        }
+
+        // ensure refresh is equal to or better than the current best video mode
+        if (vm.refresh < bestVideoMode.refresh) {
+            continue;
+        }
+
+        // save the best video mode
+        bestVideoMode = vm;
+    }
+
+    debugPrint("Available video modes:\n");
+    for (VIDEO_MODE vm : availableVideoModes) {
+        debugPrint("Width: %d, Height: %d, BPP: %d, Refresh: %d\n", vm.width, vm.height, vm.bpp, vm.refresh);
+    }
+    Sleep(2000);
+
+    // set the best video mode
+    XVideoSetMode(bestVideoMode.width, bestVideoMode.height, bestVideoMode.bpp, bestVideoMode.refresh);
+
+    debugPrint("Best video mode:\n");
+    debugPrint("Width: %d, Height: %d, BPP: %d, Refresh: %d\n", bestVideoMode.width, bestVideoMode.height, bestVideoMode.bpp, bestVideoMode.refresh);
+
+    while (1) {
+        Sleep(1000);
```

Comment:
For a sample, it would be nice to allow the user to switch between multiple video modes, maybe with some visuals that clearly demonstrate the quality difference

Reply:
Would using a png/jpg be okay for the visuals?

Reply:
For Matt's suggestion, I feel like the user's video selection should go into the "overall" display sample of 2D and 3D render with basic stats info such as how much ram is in use and frames per second. It would be neat to use abaire's 3D objects for misc interaction stuff, such as changing, moving, and rotating the object. At this time, it might be better to go with the basic draw render in 2D. With the interactive video mode selection of course.

Reply:
fyi, we already have a sample with a 3d object: mesh